### PR TITLE
Playwright POC

### PR DESCRIPTION
**Pull request by touilleMan**
Nothing fancy, move along

run the test with:
```sh
$ npm run pw:e2e:headless

> client-vite@0.0.0 pw:e2e:headless
> npx playwright test --reporter=null --timeout=10000


Running 1 test using 1 worker
[chromium] › about_modal.spec.ts:30:1 › Opens the settings dialog
>  [vite] connecting...
>  [vite] connected.
>  You are running the esm-bundler build of vue-i18n. It is recommended to configure your bundler to explicitly replace feature flag globals with boolean literals to get proper tree-shaking in the final bundle.
>  window.nextStageHook coolorg parsec://127.0.0.1:6770?no_ssl=true
  1 passed (1.4s)
```

### CODE REVIEW

**Review by NicoTuxx — APPROVED** [review]

File: client/playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+export default defineConfig({
+  testDir: './tests/pw/e2e',
+  /* Run tests in files in parallel */
+  fullyParallel: true,
+  /* Fail the build on CI if you accidentally left test.only in the source code. */
+  forbidOnly: !!process.env.CI,
+  /* Retry on CI only */
+  retries: process.env.CI ? 2 : 0,
+  /* Opt out of parallel tests on CI. */
+  workers: process.env.CI ? 1 : undefined,
```
Comment (FirelightFlagboy) [inline client/playwright.config.ts:23]:
```suggestion
  workers: undefined,
```
A specific reason to limit the number of worker on the CI ?

File: client/playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
+  reporter: 'html',
+  webServer: {
+    command: 'npm run dev',
+    url: 'http://localhost:8080',
+    reuseExistingServer: true,
+  },
```
Comment (FirelightFlagboy) [inline client/playwright.config.ts:30]: This is configured 2 times (see end of file)

**Review by FirelightFlagboy — CHANGES_REQUESTED** [review]

File: client/src/services/storageManager.ts [hunk 3]
```diff
@@ -115,7 +115,10 @@ export class StorageManager {
     const data = await this.internalStore.get(key);
 
     try {
-      const parsedData = JSON.parse(data);
+      let parsedData = JSON.parse(data);
+      if (!parsedData) {
+        parsedData = {};
+      }
```
Comment (FirelightFlagboy) [inline client/src/services/storageManager.ts:121]:
```suggestion
      const parsedData = JSON.parse(data) || {};
```

File: client/playwright.config.ts (hunk 1 above)
Comment (Max-7) [reply client/playwright.config.ts]: No idea, I removed it.

**Review by FirelightFlagboy — APPROVED** [review]

File: .github/workflows/ci-web.yml [hunk 4]
```diff
@@ -144,3 +144,15 @@ jobs:
           TESTBED_SERVER_URL: parsec3://localhost:6777?no_ssl=true
         working-directory: client
         timeout-minutes: 20
+
+      - name: Install Playwright dependencies
+        run: npx playwright install
```
Comment (FirelightFlagboy) [inline .github/workflows/ci-web.yml:149]:
```suggestion
        run: npx playwright install --with-deps
```

https://playwright.dev/docs/ci-intro#on-pushpull_request

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**